SDL2pp/Surface.cc

```diff
@@ -41,16 +41,33 @@ Surface::Surface(SDL_Surface* surface) : surface_(surface) {
 	assert(surface);
 }
 
-Surface::Surface(Uint32 flags, int width, int height, int depth, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask) {
-	if ((surface_ = SDL_CreateRGBSurface(flags, width, height, depth, Rmask, Gmask, Bmask, Amask)) == nullptr)
+Surface::Surface(int width, int height, int depth, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask) {
+	// Flags are obsolete and should be set to 0 according to the documentation
+	if ((surface_ = SDL_CreateRGBSurface(0, width, height, depth, Rmask, Gmask, Bmask, Amask)) == nullptr) {
 		throw Exception("SDL_CreateRGBSurface");
+	}
 }
 
 Surface::Surface(void* pixels, int width, int height, int depth, int pitch, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask) {
 	if ((surface_ = SDL_CreateRGBSurfaceFrom(pixels, width, height, depth, pitch, Rmask, Gmask, Bmask, Amask)) == nullptr)
 		throw Exception("SDL_CreateRGBSurfaceFrom");
 }
 
+#if SDL_VERSION_ATLEAST(2, 0, 5)
+Surface::Surface(int width, int height, int depth, Uint32 format) {
```

Comment:
Here SDL docs **do** say it's obsolete, but it really looks strange, since SDL_CreateRGBSurfaceWithFormat was just introduced. Again, I'd stick with upstream API, for consistency, ease of migration and avoiding potential problems in future.